docs/source/playbook.rst

```diff
@@ -0,0 +1,191 @@
+
+* localhost
+* packages
+
+As with regular Ansible, `localhost` is used to operate on the local machine. This is typically used to setup or work on environments.
+
+Packages is the entire set of all packages. These are exposed on the command line to users so they can operate on a limited set of packages.
+
+Note that hosts must always be a list. Our setup playbook is an example of a local connection:
+
+.. literalinclude:: ../../obal/data/playbooks/setup/playbook.yaml
+  :language: yaml
+  :caption: setup.yaml
+  :emphasize-lines: 2,3
+
+When dealing with packages we typically include the `package_variables` role to set various variables:
+
+.. literalinclude:: ../../obal/data/playbooks/changelog/playbook.yaml
+  :language: yaml
+  :caption: playbook.yaml
+  :emphasize-lines: 6,7
+
+Exposing playbooks using metadata
+---------------------------------
+
+By default Obal exposes a playbook based on its name. It can also automatically detect whether it accepts a packages parameter. To provide a better experience we introduce metadata via `metadata.yaml` in the same directory.
+
+An example:
+
+.. literalinclude:: ../../tests/fixtures/playbooks/dummy/metadata.yaml
+  :language: yaml
+  :caption: playbooks/dummy/metadata.yaml
+
+The help text is a top level key. The first line is used in ``obal --help``:
+
+.. code-block:: none
+
+   usage: obal [-h] action ...
+
+   positional arguments:
+     action          which action to execute
+       dummy         Short description
+
+   optional arguments:
+     -h, --help      show this help message and exit
+
+
+When we execute ``obal dummy --help`` we see more show up:
+
+.. code-block:: none
+
+    usage: obal dummy [-h] [-v] [-e EXTRA_VARS] [--step] [-t TAGS]
+                  [--skip-tags SKIP_TAGS] [--automatic AUTOMATIC]
+                  [--explicit MAPPED]
+                  package [package ...]
+
+   Short description
+
+   Full text on multiple lines
+   with an explicit newline
+
+   positional arguments:
+     package               the package to build
+
+   optional arguments:
+     -h, --help            show this help message and exit
+     -v, --verbose         verbose output
+     --automatic AUTOMATIC
+                           Automatically determined parameter
+     --explicit MAPPED     Explicitly specified parameter
+
+   advanced arguments:
+     -e EXTRA_VARS, --extra-vars EXTRA_VARS
+                           set additional variables as key=value or YAML/JSON, if
+                           filename prepend with @
+     --step                interactive: confirm each task before running
+     -t TAGS, --tags TAGS  only run plays and tasks tagged with these values
+     --skip-tags SKIP_TAGS
+                           only run plays and tasks whose tags do not match these
+                           values
+
+
+Help
+~~~~
+
+Help is a string at the top level in the metadata. It's a string with some newline handling.
```

Comment:
Could drop the somewhat redundant `it's a string` and combine this into one sentence.